app/models/node.rb

```diff
@@ -4,7 +4,7 @@ class Node < ApplicationRecord
   validates :name, presence: true
   validates :parent, presence: true, if: :parent_id
   validates_presence_of :description, :allow_blank => true
-  validates :cost_code, :presence => true, :allow_blank => true, format: { with: /\AS[0-9]{4}+\z/, message: 'must be of the format "S" followed by four digits' }, :on => [:create, :update]
+  validates :cost_code, :presence => true, :allow_blank => true, format: { with: /\AS[\d]{4}(_[\d]{1,2}){0,1}\z/, message: 'must be a valid project or subproject cost code' }, :on => [:create, :update]
```

Comment:
Better to use [0-9] rather than \d. The \d group contains all sorts of weird characters, but we are only looking for normal decimal digits 0-9.